### PR TITLE
Data package build

### PR DESCRIPTION
**Pull request by blaiszik**

### CODE REVIEW

File: foundry/foundry.py [hunk 1]
```diff
@@ -498,3 +512,41 @@ def download_file(file):
...
+
+        if file:
+            with open(file, "r") as fp:
+                fs = FoundrySpecification(**json.load(fp))
+        else:
+            fs = FoundrySpecification(**spec)
+
+        fs.remove_duplicate_dependencies()
+
+        results = Parallel(n_jobs=num_cores)(
```
Comment (ascourtas) [inline foundry/foundry.py:546]: what does this do? can't tell from context, could you please add an inline comment?

File: foundry/foundry.py [hunk 2]
```diff
@@ -498,3 +512,41 @@ def download_file(file):
...
+        Returns
+        -------
+        (Foundry): self: for chaining
+        """
+
+        print("Building Data Package")
+        num_cores = multiprocessing.cpu_count()
+
+        def start_download(ds, interval=interval, globus=False):
+            print("=== Fetching Data Package {} ===".format(ds.name))
```
Comment (ascourtas) [inline foundry/foundry.py:533]: I like the progress print statements, v helpful!

File: foundry/foundry.py [hunk 3]
```diff
@@ -498,3 +512,41 @@ def download_file(file):
             print(results)
 
         return self
+
+    def build(self, spec, globus=False, interval=3, file=False):
+        """Build a Foundry Data Package
```
Comment (ascourtas) [inline foundry/foundry.py:517]: Could you add a little more clarification in the docstring on how this is used? Looks like it instantiates the foundry data package specification but I'm not entirely sure how that would be used.

Comment (ascourtas) [reply foundry/foundry.py:517]: after spending some more time on this function, it looks like it actually downloads the data package per the FoundrySpec? is that right?

**Review by ascourtas — CHANGES_REQUESTED** [review]
just a few clarification requests on `build()`, for the most part. weird that some of the edits are exactly the same as in the previous PR we just merged, should be ok though

File: foundry/foundry.py [hunk 4]
```diff
@@ -293,7 +307,7 @@ def publish(self, foundry_metadata, data_source, title, authors, update=False, p
             affiliations=kwargs.get("affiliations", []),
             subjects=kwargs.get("tags", ["machine learning", "foundry"]),
             publisher=kwargs.get("publisher", ""),
-            publication_year=publication_year
+            publication_year=publication_year,
```
[the remark this answers is not in the document]
Comment (blaiszik) [reply foundry/foundry.py:310]: This is an auto formatting thing from Black formatter. I'm not sure exactly why it does this, but it does it automatically...

[the remark this answers is not in the document]
Comment (ascourtas) [reply foundry/foundry.py:310]: I googled it, somehow it's supposed to be useful when diffing files in version control? https://www.python.org/dev/peps/pep-0008/#when-to-use-trailing-commas

[the remark this answers is not in the document]
Comment (ascourtas) [reply foundry/foundry.py:310]: I'm fine with leaving it!

**Pull request merged**